src/Components/ServiceSelectionScene/AddLabelToFiltersHeaderActionScene.tsx

```diff
@@ -0,0 +1,117 @@
+import { SceneComponentProps, SceneObjectBase, SceneObjectState } from '@grafana/scenes';
+import { Button, useStyles2 } from '@grafana/ui';
+import { testIds } from '../../services/testIds';
+import { addToFilters, FilterType } from '../ServiceScene/Breakdowns/AddToFiltersButton';
+import { VAR_LABELS } from '../../services/variables';
+import { getLabelsVariable, getValueFromAdHocVariableFilter } from '../../services/variableGetters';
+import { reportAppInteraction, USER_EVENTS_ACTIONS, USER_EVENTS_PAGES } from '../../services/analytics';
+import { css } from '@emotion/css';
+import React from 'react';
+import { FilterOp } from '../../services/filterTypes';
+
+export interface AddLabelToFiltersHeaderActionSceneState extends SceneObjectState {
+  name: string;
+  value: string;
+  isIncluded?: boolean;
+  isExcluded?: boolean;
+}
+
+export class AddLabelToFiltersHeaderActionScene extends SceneObjectBase<AddLabelToFiltersHeaderActionSceneState> {
+  constructor(state: AddLabelToFiltersHeaderActionSceneState) {
+    super({
+      ...state,
+    });
+
+    this.addActivationHandler(this.onActivate.bind(this));
+  }
+
+  onActivate() {
+    this.setState({ ...this.isSelected() });
+    this._subs.add(
+      getLabelsVariable(this).subscribeToState(() => {
+        const selected = this.isSelected();
+        if (this.state.isIncluded !== selected.isIncluded || this.state.isExcluded !== selected.isExcluded) {
+          this.setState({ ...selected });
+        }
+      })
+    );
+  }
+
+  isSelected = () => {
+    const variable = getLabelsVariable(this);
+
+    // Check if the filter is already there
+    const filterInSelectedFilters = variable.state.filters.find((f) => {
+      const value = getValueFromAdHocVariableFilter(variable, f);
+      return f.key === this.state.name && value.value === this.state.value;
+    });
+
+    if (!filterInSelectedFilters) {
+      return { isIncluded: false, isExcluded: false };
+    }
+
+    return {
+      isIncluded: filterInSelectedFilters.operator === FilterOp.Equal,
```

Comment:
Can something be included and excluded at the same time? If not, I would recommend changing this to a boolean or enum.
Can also be `included: boolean | null`. null === no state, included true, excluded false.